### PR TITLE
API for obtaining global gradient norm

### PR DESCRIPTION
**Pull request by tjruwase**
API for obtaining global unclipped gradient norm across all parameters groups.  Based off #1286.
Optimizers are solely responsible for computing gradient norms. Gradient norms are computed (or refreshed) in optimizer.step().

@stas00 FYI

### CODE REVIEW

**Comment by stas00** [comment]
Thank you for working on this, @tjruwase 

and then we will need Shaden's https://github.com/microsoft/Megatron-DeepSpeed/pull/8 on the Megatron side once this is merged.

**Review by ShadenSmith — APPROVED** [review]
@samyam may also want to take a look
